sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java

```diff
@@ -0,0 +1,225 @@
+        mApplicationLifecycleListener.registerApplicationLifecycleCallbacks(mockAppCenterService1);
+        mApplicationLifecycleListener.registerApplicationLifecycleCallbacks(mockAppCenterService2);
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(mockAppCenterService1).onApplicationEnterForeground();
+        verify(mockAppCenterService2).onApplicationEnterForeground();
+
+        /* Call onActivityResumed. */
+        mApplicationLifecycleListener.onActivityResumed(mActivityMock);
+        verify(mHandlerMock, never()).removeCallbacks(any(Runnable.class));
+
+        /* Call onActivityStopped. */
+        mApplicationLifecycleListener.onActivityStopped(mActivityMock);
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(mockAppCenterService1).onApplicationEnterForeground();
+        verify(mockAppCenterService2).onApplicationEnterForeground();
+    }
+
+    @Test
+    public void onExpectedLifecycleTest() {
+
+        /* Prepare data. */
+        Bundle mockBundle = mock(Bundle.class);
+        MockCallbacks mockAppCenterService1 = spy(new MockCallbacks());
+        MockCallbacks mockAppCenterService2 = spy(new MockCallbacks());
+        mApplicationLifecycleListener.registerApplicationLifecycleCallbacks(mockAppCenterService1);
+        mApplicationLifecycleListener.registerApplicationLifecycleCallbacks(mockAppCenterService2);
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityCreated(mActivityMock, mockBundle);
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(mockAppCenterService1).onApplicationEnterForeground();
+        verify(mockAppCenterService2).onApplicationEnterForeground();
+
+        /* Call onActivityResumed. */
+        mApplicationLifecycleListener.onActivityResumed(mActivityMock);
+        verify(mHandlerMock, never()).removeCallbacks(any(Runnable.class));
+        mApplicationLifecycleListener.onActivitySaveInstanceState(mActivityMock, mockBundle);
+
+        /* Call onActivityPaused. */
+        mApplicationLifecycleListener.onActivityPaused(mActivityMock);
+        verify(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+
+        /* Call onActivityStopped. */
+        mApplicationLifecycleListener.onActivityStopped(mActivityMock);
+
+        /* Call onActivityDestroyed. */
+        mApplicationLifecycleListener.onActivityDestroyed(mActivityMock);
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(mockAppCenterService1, times(2)).onApplicationEnterForeground();
+        verify(mockAppCenterService2, times(2)).onApplicationEnterForeground();
+    }
+
+    private class MockCallbacks extends AbstractAppCenterService implements ApplicationLifecycleListener.ApplicationLifecycleCallbacks {
```

Comment:
```suggestion
    private class MockService extends AbstractAppCenterService {
```

I guess we already have some mock service for other tests.. Can we reuse it?